cf-oscore/src/main/java/org/eclipse/californium/oscore/OSCoreEndpointContextInfo.java

```diff
@@ -0,0 +1,207 @@
+     *
+     * This new endpoint context is available using
+     * MessageObserver.onContextEstablished(EndpointContext).
+     *
+     */
+    private static class OSCoreEndpointContextOperator implements EndpointContextOperator {
+
+        /**
+         * OSCORE context to take information from to set in the endpoint context
+         */
+        private final OSCoreCtx oscoreCtx;
+
+        /**
+         * Constructor taking an OSCORE Context that the information to be set
+         * in the endpoint context will be taken from.
+         *
+         * @param oscoreCtx the OSCORE context to take information to set from
+         */
+        public OSCoreEndpointContextOperator(final OSCoreCtx oscoreCtx) {
+            this.oscoreCtx = oscoreCtx;
+        }
+
+        /**
+         * Method that will be ran when setting the endpoint context for an exchange.
+         * Should perform changes to the endpoint context to add the OSCORE-related strings.
+         *
+         * @return the new endpoint context to use
+         */
+        @Override
+        public EndpointContext apply(EndpointContext context) {
+            return setInfo(oscoreCtx, context);
+        }
+
+    }
+
+    /**
+     * Sets information in a source endpoint context for incoming requests.
+     *
+     * @param oscoreCtx the OSCORE context used for this message
+     * @param request the request to set the endpoint context for
+     */
+    public static void receivingRequest(OSCoreCtx oscoreCtx, Request request) {
+        setInfoIncoming(oscoreCtx, request);
+    }
+
+    /**
+     * Sets information in a destination endpoint context for outgoing responses.
+     *
+     * @param oscoreCtx the OSCORE context used for this message
+     * @param response the response to set the endpoint context for
+     */
+    public static void sendingResponse(OSCoreCtx oscoreCtx, Response response) {
+        setInfoOutgoing(oscoreCtx, response);
+    }
+
+    /**
+     * Sets information in a destination endpoint context for incoming responses.
+     *
+     * @param oscoreCtx the OSCORE context used for this message
+     * @param response the response to set the endpoint context for
+     */
+    public static void receivingResponse(OSCoreCtx oscoreCtx, Response response) {
+        setInfoIncoming(oscoreCtx, response);
+    }
+
+    /**
+     * Adds strings with information about the OSCORE context used to this
+     * destination endpoint context for outgoing messages.
+     *
+     * @param oscoreCtx the OSCORE context used for this message
+     * @param message the message to set information in the destination endpoint context for
+     */
+    private static void setInfoOutgoing(OSCoreCtx oscoreCtx, Message message) {
+
+        //Do nothing if this is not an instance of MapBasedEndpointContext (must cast to it)
+        if (message.getDestinationContext() instanceof MapBasedEndpointContext == false) {
```

Comment:
Please, don't introduce a dependency on `MapBasedEndpointContext` instances. 
That dependency is not required!

Reply:
This should be fixed in commit d0c4594169ec6a56545192005ef395b2f69cf501